### PR TITLE
Update to v2.2.28

### LINKED ISSUE #44: udunits 2.2.28 was released 2020-12-28 deprecates 2.2.27.27
The [latest `udunits` source is 2.2.28](https://www.unidata.ucar.edu/software/udunits/CHANGE_LOG), whereas [2.2.27.27 is the latest version available through conda-forge](https://anaconda.org/conda-forge/udunits2/files).

It'd be great to update this to the latest release.

### PR DESCRIPTION
**Pull request by xylar**
To do this, we need to change the source location.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #44

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+{% set version = "2.2.28" %}
 
 package:
   name: udunits2
   version: {{ version }}
 
 source:
-  url: https://github.com/Unidata/UDUNITS-2/archive/v{{ version }}.tar.gz
-  sha256: 6c65b3097ef38b94c01dfede8904346c1439de9704b9bae7deea72bd5143a7bb
+  url: https://artifacts.unidata.ucar.edu/repository/downloads-udunits/{{ version }}/udunits-{{ version }}.tar.gz
```
Comment (xylar) [inline recipe/meta.yaml:8]: @conda-forge/udunits2, it seems that the 2.2.28 tag was never created on GitHub.  Thus the change in location for the source.  We can change back if tagging happens as expected for future releases.  See:
https://github.com/Unidata/UDUNITS-2/issues/104
and
https://github.com/Unidata/UDUNITS-2/issues/109

Comment (dopplershift) [reply recipe/meta.yaml:8]: I would consider the new location to be the canonical location.

Comment (xylar) [reply recipe/meta.yaml:8]: Hopefully, the bot will understand it well enough to notice new releases.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
   name: udunits2
   version: {{ version }}
 
 source:
-  url: https://github.com/Unidata/UDUNITS-2/archive/v{{ version }}.tar.gz
-  sha256: 6c65b3097ef38b94c01dfede8904346c1439de9704b9bae7deea72bd5143a7bb
+  url: https://artifacts.unidata.ucar.edu/repository/downloads-udunits/{{ version }}/udunits-{{ version }}.tar.gz  # [not win]
+  sha256: 590baec83161a3fd62c00efa66f6113cec8a7c461e3f61a5182167e0cc5d579e  # [not win]
+  url: https://artifacts.unidata.ucar.edu/repository/downloads-udunits/{{ version }}/udunits-{{ version }}.zip  # [win]
+  sha256: e09d31db68f9a840a0663c7e9909101957733ff0310761b9906f4722e0d92c44  # [win]
```
Comment (xylar) [inline recipe/meta.yaml:11]: There appear to have been some pretty significant packaging problems in which necessary files for CMake were included in the `.zip` file but not in the `.tar.gz` file.  Hopefully, this selector is only necessary for this one build.  See https://github.com/Unidata/UDUNITS-2/issues/108 for more on this.

**Comment by xylar** [comment]
@conda-forge/udunits2, this is ready to review.  Feel free to merge it, I just want another pair of eyes on it.

**Pull request merged**